src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/AddAzureVMDataDiskCommand.cs

```diff
@@ -12,148 +12,259 @@
+    public class AddAzureVMDataDiskCommand : ComputeClientBaseCmdlet
     {
+        protected const string VmNormalDiskParameterSet = "VmNormalDiskParameterSetName";
+        protected const string VmManagedDiskParameterSet = "VmManagedDiskParameterSetName";
+        protected const string VmScaleSetVMParameterSet = "VmScaleSetVMParameterSetName";
+
         [Alias("VMProfile")]
         [Parameter(
+            ParameterSetName = VmNormalDiskParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMProfile)]
+        [Parameter(
+            ParameterSetName = VmManagedDiskParameterSet,
             Mandatory = true,
             Position = 0,
             ValueFromPipeline = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = HelpMessages.VMProfile)]
         [ValidateNotNullOrEmpty]
-        public PSVirtualMachine VM { get; set; }
+        public CM.PSVirtualMachine VM { get; set; }
 
         [Parameter(
-            Mandatory = false,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = VmScaleSetVMParameterSet,
+            HelpMessage = HelpMessages.VmssVMProfile)]
+        [ValidateNotNullOrEmpty]
+        public PSVirtualMachineScaleSetVM VirtualMachineScaleSetVM { get; set; }
+
+        [Parameter(
+            Mandatory = true,
             Position = 1,
             ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskLun)]
+        [ValidateNotNullOrEmpty]
+        public int? Lun { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskCreateOption)]
+        public string CreateOption { get; set; }
+
+        [Parameter(
+            ParameterSetName = VmNormalDiskParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskName)]
+        [Parameter(
+            ParameterSetName = VmManagedDiskParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
             HelpMessage = HelpMessages.VMDataDiskName)]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         [Parameter(
             Mandatory = false,
-            Position = 2,
+            ParameterSetName = VmNormalDiskParameterSet,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = HelpMessages.VMDataDiskVhdUri)]
         [ValidateNotNullOrEmpty]
         public string VhdUri { get; set; }
 
         [Parameter(
             Mandatory = false,
-            Position = 3,
```

Comment:
@hyonholee any reason that positions are being removed from parameters in this file? This would be a breaking change

Reply:
@cormacpayne  this is optional parameter, so I removed the position value

Reply:
@hyonholee I agree that this is correct pattern, but it's still a breaking change. We should look at making this kind of change across all Compute cmdlets in the next breaking change release